src/NServiceBus.Core/Pipeline/Incoming/DeserializeLogicalMessagesConnector.cs

```diff
@@ -31,7 +31,12 @@ public DeserializeLogicalMessagesConnector(MessageDeserializerResolver deseriali
             }
         }
 
-        List<LogicalMessage> ExtractWithExceptionHandling(IncomingMessage message)
+        static bool IsControlMessage(IncomingMessage incomingMessage)
+        {
+            return incomingMessage.Headers.ContainsKey(Headers.ControlMessageHeader) && incomingMessage.Headers[Headers.ControlMessageHeader] == true.ToString();
```

Comment:
Just verified again that `true.ToString()` is equivalent to `Boolean.TrueString`. It uses the literals internally and ignores any culture specific settings. This is really confusing and always trips me over

http://referencesource.microsoft.com/#mscorlib/system/boolean.cs,91